issue #3 Add multilingual support

csKyou opened the pull request.

csKyou committed:
issue #3 Add multilingual support

dixonsiu commented on src/LLMessage.html:
```diff
@@ -20,30 +24,34 @@
     </a>
     -->
-        <h1>わかば市</h1>
+        <h1 data-i18n="llmessage:header"></h1>
```
Please avoid using generic naming here.

Suggestion: llmessage:wakabaCity

csKyou replied:
Changed the sentence to a predictable name.

dixonsiu commented on src/LLMessage.html:
```diff
@@ -20,30 +24,34 @@
     </section>
-    <a href="" class="list-readmore">もっとみる</a>
+    <a href="" class="list-readmore" data-i18n="llmessage:more"></a>
+    <section class="login-section">
```
1. Delete the class attribute if not needed, OR,
2. Rename the class and the corresponding CSS style

csKyou replied:
Deleted the class attribute.

dixonsiu commented on src/js/common.js:
```diff
@@ -10,6 +10,38 @@ Common.refExpires = sessionStorage.getItem("ISRefExpires");
+    $('[data-i18n]').localize();
+}
+
+Common.AddResourceProfile = function(lng , ns, id, json) {
```
This implementation only works now, it will not work properly if lng is not "en" or "ja". Therefore, it is better to use only line 35 and 41 for now. Besides, profile.json does not support multi-languages currently.

csKyou replied:
Deleted the processing of i18n and restored it to use the profile as it is.

dixonsiu commented on src/js/LLMessage.js:
```diff
@@ -248,13 +277,17 @@ lm.getReceiveMessage = function() {
                         var targetImage = prof[0].Image;
-                        var targetName = prof[0].DisplayName
+                        //var targetName = prof[0].DisplayName
+                        var targetName = "llmessage:approved" + i;
```
Remove i18next and just simply display  prof[0].DisplayName + prof[0].Description.

csKyou replied:
Deleted the processing of i18n and restored it to use the profile as it is.

dixonsiu commented on src/js/LLMessage.js:
```diff
@@ -314,7 +347,9 @@ lm.getReceiveMessage = function() {
-                if (lm.isExpired(messageBody.TermEnd)) {
+
+                // If you want to display the message of the past day, comment out the if statement.
+                //if (lm.isExpired(messageBody.TermEnd)) {
```
Delete it if you think you don't need this feature in the near future. OR,
Implement it as an option.
old:

    if (lm.isExpired(messageBody.TermEnd)) {

suggested:

    lm.showExpiredMessage = true;
    ...
    if (lm.showExpiredMessage || lm.isExpired(messageBody.TermEnd)) {

csKyou replied:
It was implemented as an option.

dixonsiu commented on src/js/LLMessage.js:
```diff
@@ -374,12 +409,16 @@ lm.setProfile = function() {
-                $('#requestName').html('<div class="sizeCaption">' + result[0].DisplayName + '</div>');
+                Common.AddResourceProfile("en", "llmessage", "msg" + i, result[0]);
+                Common.AddResourceProfile("ja", "llmessage", "msg" + i, result[0]);
+                $('#requestName').html('<div class="sizeCaption" data-i18n="llmessage:msg' + i + '_DisplayName"></div>').localize();
```
Revert back to single language

csKyou replied:
Deleted the processing of i18n and restored it to use the profile as it is.

dixonsiu commented on src/js/LLMessage.js:
```diff
@@ -413,6 +452,9 @@ lm.moveReplay = function(id, numerator) {
 
+/*
+ * If the received date is past in the past, we will return false.
+ */
```
Comments: If the retrieved message's timestamp is not today, it is considered expired.

[Timestamp issue](https://github.com/personium/personium-core/issues/42)

csKyou replied:
It has changed.

csKyou commented on src/js/LLMessage.js:
```diff
@@ -413,6 +452,9 @@ lm.moveReplay = function(id, numerator) {
 
+/*
+ * If the received date is past in the past, we will return false.
+ */
```
It has changed.

csKyou commented on src/js/common.js:
```diff
@@ -10,6 +10,38 @@ Common.refExpires = sessionStorage.getItem("ISRefExpires");
+    $('[data-i18n]').localize();
+}
+
+Common.AddResourceProfile = function(lng , ns, id, json) {
```
Fix to use the result of profile.json as it is.

csKyou commented on src/LLMessage.html:
```diff
@@ -20,30 +24,34 @@
     </a>
     -->
-        <h1>わかば市</h1>
+        <h1 data-i18n="llmessage:header"></h1>
```
Changed the sentence to a predictable name.

csKyou committed:
issue #3 Add multilingual support
 - Reflect review results
 - We corresponded because there was a file which multi-language correspondence was missing.
